### ISSUE TITLE
failed to run custom build command for `nexus-network v0.4.0

### ISSUE THREAD

**Issue by LayVilla05** [nexus-xyz/nexus-cli#133]
Compiling nexus-network v0.4.0 (C:\Users\layvi\.nexus\network-api\clients\cli)
The following warnings were emitted during compilation:

warning: nexus-network@0.4.0: Failed to run protoc: program not found

error: failed to run custom build command for `nexus-network v0.4.0 (C:\Users\layvi\.nexus\network-api\clients\cli)`

Caused by:
  process didn't exit successfully: `C:\Users\layvi\.nexus\network-api\clients\cli\target\release\build\nexus-network-a3fffeb85525b117\build-script-build` (exit code: 1)
  --- stdout
  cargo:warning=Failed to run protoc: program not found

  --- stderr
  Error: Error { kind: NotFound, message: "program not found" }
curl: (6) Could not resolve host: sh

**Comment by Ahnetdd** [comment]
My server is running on linux, ubuntu 22.04. I have the same problem. What could be the solution to the problem?

**Comment by Mantodkaz** [comment]
> My server is running on linux, ubuntu 22.04. I have the same problem. What could be the solution to the problem?

have you installed this package before?
apt install protobuf-compiler -y

**Comment by Ahnetdd** [comment]
Yes. After installing this package the problem is solved. Thanks for the reply.

**Comment by Giangr2812** [comment]
> > My server is running on linux, ubuntu 22.04. I have the same problem. What could be the solution to the problem?
> 
> have you installed this package before? apt install protobuf-compiler -y
> 
> If it still doesn't work, Install New Version of Protoc Manually curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-linux-x86_64.zip unzip protoc-21.12-linux-x86_64.zip -d /usr/local/protoc echo 'export PATH=/usr/local/protoc/bin:$PATH' >> ~/.bashrc source ~/.bashrc
> 
> try it, and run again. good luck

I'm using macos sonoma 14.7 and it has an error that doesn't work

**Comment by Mantodkaz** [comment]
> > > My server is running on linux, ubuntu 22.04. I have the same problem. What could be the solution to the problem?
> > 
> > 
> > have you installed this package before? apt install protobuf-compiler -y
> > If it still doesn't work, Install New Version of Protoc Manually curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-linux-x86_64.zip unzip protoc-21.12-linux-x86_64.zip -d /usr/local/protoc echo 'export PATH=/usr/local/protoc/bin:$PATH' >> ~/.bashrc source ~/.bashrc
> > try it, and run again. good luck
> 
> I'm using macos sonoma 14.7 and it has an error that doesn't work

for macos you can try use this version
curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-osx-x86_64.zip
unzip protoc-21.12-osx-x86_64.zip -d /usr/local/protoc
echo 'export PATH=/usr/local/protoc/bin:$PATH' >> ~/.zshrc
source ~/.zshrc
chmod +x /usr/local/protoc/bin/protoc

**Comment by Giangr2812** [comment]
> > > > My server is running on linux, ubuntu 22.04. I have the same problem. What could be the solution to the problem?
> > > 
> > > 
> > > have you installed this package before? apt install protobuf-compiler -y
> > > If it still doesn't work, Install New Version of Protoc Manually curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-linux-x86_64.zip unzip protoc-21.12-linux-x86_64.zip -d /usr/local/protoc echo 'export PATH=/usr/local/protoc/bin:$PATH' >> ~/.bashrc source ~/.bashrc
> > > try it, and run again. good luck
> > 
> > 
> > I'm using macos sonoma 14.7 and it has an error that doesn't work
> 
> for macos you can try use this version curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-osx-x86_64.zip unzip protoc-21.12-osx-x86_64.zip -d /usr/local/protoc echo 'export PATH=/usr/local/protoc/bin:$PATH' >> ~/.zshrc source ~/.zshrc chmod +x /usr/local/protoc/bin/protoc
tôi đang chạy và bị lỗi như vậy
Archive:  protoc-21.12-osx-x86_64.zip
checkdir:  cannot create extraction directory: /usr/local/protoc
           Permission denied

**Comment by Giangr2812** [comment]
> > > > My server is running on linux, ubuntu 22.04. I have the same problem. What could be the solution to the problem?
> > > 
> > > 
> > > have you installed this package before? apt install protobuf-compiler -y
> > > If it still doesn't work, Install New Version of Protoc Manually curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-linux-x86_64.zip unzip protoc-21.12-linux-x86_64.zip -d /usr/local/protoc echo 'export PATH=/usr/local/protoc/bin:$PATH' >> ~/.bashrc source ~/.bashrc
> > > try it, and run again. good luck
> > 
> > 
> > I'm using macos sonoma 14.7 and it has an error that doesn't work
> 
> for macos you can try use this version curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-osx-x86_64.zip unzip protoc-21.12-osx-x86_64.zip -d /usr/local/protoc echo 'export PATH=/usr/local/protoc/bin:$PATH' >> ~/.zshrc source ~/.zshrc chmod +x /usr/local/protoc/bin/protoc

i am running and it shows error like this
![image](https://github.com/user-attachments/assets/1a71b583-4025-4121-bd69-1a6f033110e9)

**Comment by mannyanebi** [comment]
> > My server is running on linux, ubuntu 22.04. I have the same problem. What could be the solution to the problem?
> 
> have you installed this package before? apt install protobuf-compiler -y
> 
> If it still doesn't work, Install New Version of Protoc Manually curl -LO [protocolbuffers/protobuf@`v21.12` protoc-21.12-linux-x86_64.zip (download)](https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-linux-x86_64.zip) unzip protoc-21.12-linux-x86_64.zip -d /usr/local/protoc echo 'export PATH=/usr/local/protoc/bin:$PATH' >> ~/.bashrc source ~/.bashrc
> 
> try it, and run again. good luck

Thanks this helped!
